### DISCUSSION TITLE
File Support Extension

### DISCUSSION THREAD

**Discussion by simwai** [imartinez/privateGPT#534] [Ideas]
Pls add support for json and python files and any other common file extension which could be in a public GitHub python project. This would make this tool so good.

**Comment by FixedAF** [comment]
could you not just add .txt to the end of those files and have it ingest? the code will sitll be there as well and the .py extension before the .txt

**Reply by simwai** [reply]
On a big GitHub project this is much work. I think it's worth to support it.

**Reply by jupurgepen** [reply]
Just add this line to ingest.py, above "# Add more mappings for other file extensions and loaders as needed:"
_".py": (TextLoader, {"encoding": "utf8"}),_

**Reply by simwai** [reply]
I think it would be good to officially support it

**Comment by FixedAF** [comment]
fair enough. Have gpt4 write you a python script to batch convert .py and .json files to .txt  "without deleting original extension from name" then paste that code into a text editor. save the file as PyJsonCnv.py (or something)  

give gpt4 the folder path to the files you want converted, the the folder path to the "shared documents" in the private GPT folder.. gpt4 will write the whole script (i been doing this trying to convert xlsx to pdf ) 
anyways after u made the script go to the folder its in with file browser (assuming ur on linux) and type "python PyJsonCnv.py" and hit enter. should spit out a log of whats happening--watch your shared documents folder populate


next step would be to put that script in the private GPT folder and have the model call upon it when needed...But i dont know how to do that--yet.

**Reply by FixedAF** [reply]
--i forgot to say open a terminal in the folder that the script u made is in, then type the python PyJson.....

**Reply by FixedAF** [reply]
import os

for file in os.listdir("."):
    if file.endswith(".py") or file.endswith(".json"):
        file = file[:-3] + ".txt"
        print(file)

save all that in a text file. change the extention to .py  



This script loops through all the files in the current directory and checks if they end with either “.py” or “.json”. If so, it changes the extension to “.txt”, which is the default extension for text files. The output of this script will be a list of all the converted files.

**Comment by jupurgepen** [comment]
You can just add this line to ingest.py, above "# Add more mappings for other file extensions and loaders as needed:"
_".py": (TextLoader, {"encoding": "utf8"}),_
